src/cast_core/include/cast_core/actors/OptionsConversion.hpp

```diff
@@ -26,216 +26,200 @@
+    using type = mongocxx::options::estimated_document_count;
+
+    static type convert(const Node& node) {
+        type rhs{};
 
-    static bool decode(const Node& node, CountOptions& rhs) {
-        if (!node.IsMap()) {
-            return false;
-        }
         if (node["MaxTime"]) {
-            auto maxTime = node["MaxTime"].as<genny::TimeSpec>();
+            auto maxTime = node["MaxTime"].to<genny::TimeSpec>();
             rhs.max_time(std::chrono::milliseconds{maxTime});
         }
         if (node["ReadPreference"]) {
-            auto readPref = node["ReadPreference"].as<mongocxx::read_preference>();
+            auto readPref = node["ReadPreference"].to<mongocxx::read_preference>();
             rhs.read_preference(readPref);
         }
-        return true;
+        return rhs;
     }
 };
 
 
 template <>
-struct convert<mongocxx::options::insert> {
-    static Node encode(const mongocxx::options::insert& rhs) {
-        return {};
-    }
-    static bool decode(const Node& node, mongocxx::options::insert& rhs) {
-        if (!node.IsMap()) {
-            return false;
-        }
+struct NodeConvert<mongocxx::options::insert> {
+    using type = mongocxx::options::insert;
+
+    static type convert(const Node& node) {
+        type rhs{};
+
         if (node["Ordered"]) {
-            rhs.ordered(node["Ordered"].as<bool>());
+            rhs.ordered(node["Ordered"].to<bool>());
         }
         if (node["BypassDocumentValidation"]) {
-            rhs.bypass_document_validation(node["BypassDocumentValidation"].as<bool>());
+            rhs.bypass_document_validation(node["BypassDocumentValidation"].to<bool>());
         }
         if (node["WriteConcern"]) {
-            rhs.write_concern(node["WriteConcern"].as<mongocxx::write_concern>());
+            rhs.write_concern(node["WriteConcern"].to<mongocxx::write_concern>());
         }
 
-        return true;
+        return rhs;
     }
 };
 
 template <>
-struct convert<mongocxx::options::transaction> {
-    using TransactionOptions = mongocxx::options::transaction;
-    static Node encode(const TransactionOptions& rhs) {
-        return {};
-    }
+struct NodeConvert<mongocxx::options::transaction> {
+    using type = mongocxx::options::transaction;
+
+    static type convert(const Node& node) {
+        type rhs{};
 
-    static bool decode(const Node& node, TransactionOptions& rhs) {
-        if (!node.IsMap()) {
-            return false;
-        }
         if (node["WriteConcern"]) {
-            auto wc = node["WriteConcern"].as<mongocxx::write_concern>();
+            auto wc = node["WriteConcern"].to<mongocxx::write_concern>();
             rhs.write_concern(wc);
         }
         if (node["ReadConcern"]) {
-            auto rc = node["ReadConcern"].as<mongocxx::read_concern>();
+            auto rc = node["ReadConcern"].to<mongocxx::read_concern>();
             rhs.read_concern(rc);
         }
+        if (node["MaxCommitTimeMs"]) {
```

Comment:
TIG-2937 new options

Reply:
Do you want to add these to the CrudActor also?

Reply:
Ah, I thought there was code in the CollectionScanner to take advantage of this also. Maybe I misread. I definitely thought I was in a different file when I added this comment.

Reply:
Looking again and I misread. Sorry and thanks.

Reply:
It got a bit of a makeover in this PR so it may have looked a bit different.

There is a lot of good stuff in CrudActor that I hope to reuse in future commits.